src/components/ContentWidgets/Homepage/HomepageEditor.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { FormEvent, ChangeEvent } from 'react';
+import { WidgetEditorProps } from '../../ContentMapping/ContentMapping';
+import './Homepage.css';
+
+export const HomepageEditor: React.FC<WidgetEditorProps> = ({
+    originalContent, 
+    editedContent,
+    setEditedContentOnChange
+}) => {
+    let homepage_content = editedContent.homepage_content || 
+        {title: "Title not set", subtitle: "Subtitle not set", backgroundImage: "", blur: "", bgLightness: "", bgOpacity: ""};
+
+    const updateTitle = (e: React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
These functions can be condensed into one function by taking advantage of the `e.target.name` property.

```
const generalUpdate = (e) => {
  homepage_content[e.target.name] = e.target.value;
  setEditedContentOnChange("homepage_content", homepage_content);
}
```

then you just add a `name="___"` property to the input component